api/src/main/java/graphql/nadel/Nadel.java

```diff
@@ -69,7 +70,8 @@
     final Map<String, StitchingDsl> stitchingDsls;
     final ServiceExecutionFactory serviceExecutionFactory;
     final List<Service> services;
-    final GraphQLSchema overallSchema;
+    final GraphQLSchema privateOverallSchema;
```

Comment:
I think we can extend  `NadelPublicGraphQLSchema` and `NadelPrivateGraphQLSchema` for better type safety (they can either extend GraphQLSchema or wrap it)